lib/ecto/adapters/postgres.ex

```diff
@@ -147,13 +147,20 @@ defmodule Ecto.Adapters.Postgres do
     end
   end
 
-  defp concat_if(content, nil, _fun),  do: content
-  defp concat_if(content, value, fun), do: content <> " " <> fun.(value)
+  defp concat_if(content, nil, _),  do: content
+  defp concat_if(content, false, _),  do: content
+  defp concat_if(content, value, fun) when is_function(fun), do: content <> " " <> fun.(value)
+  defp concat_if(content, _value, append), do: content <> " " <> append
 
   @impl true
   def storage_down(opts) do
     database = Keyword.fetch!(opts, :database) || raise ":database is nil in repository configuration"
-    command  = "DROP DATABASE \"#{database}\""
+    force = case Keyword.fetch(opts, :force_drop) do
```

Comment:
Should we simply call this option `--force`? We also need to change `mix ecto.drop` to make sure it accepts the `--force` option. Or are you actually setting this in your repo configuration? 🤔

Reply:
I was setting it in the Repo config but it's up for debate. I think it makes more sense as param on `mix ecto.drop`. Will fix.

Reply:
There is already a `--force` option on `mix ecto.drop`:
```
    * `-f`, `--force` - do not ask for confirmation when dropping the database.
      Configuration is asked only when `:start_permanent` is set to true
      (typically in production)
```

What do to?

Reply:
@ruudk I think we can reuse the option but I am not sure. What do you think @wojtekmach?

Reply:
I assume `WITH FORCE` is only available on pg 13+ which means it's gonna fail on earlier versions? If so, we need to add another option to not break existing users.

Let's add a `--force-drop` option to `mix ecto.drop` and perhaps say that it's only used by adapters supporting it.

Reply:
What if we just first determine the PostgreSQL version by doing:
```sql
SHOW server_version_num;
# 130000
```

And then use it to apply the `WITH (FORCE)` when `version >= 130_000`

Reply:
Maybe we can write this logic conditionally in Postgres itself, using a single command? Otherwise I would have a separate flag as I would like to avoid doing multiple operations.

Reply:
I tried this:
```sql
               DO $$
               DECLARE pg_version int := current_setting('server_version_num');
               BEGIN
                 IF pg_version >= 130000 THEN
                   DROP DATABASE "#{database}" WITH (FORCE);
                 ELSE
                   DROP DATABASE "#{database}";
                 END IF;
               END $$ LANGUAGE plpgsql;
```

But it doesn't work because you cannot drop a database inside a transaction.

```
ERROR 25001 (active_sql_transaction) DROP DATABASE cannot be executed from a function
```

Reply:
Yeah, so a separate flag, like --force-drop as you originally proposed is def. the best way to go IMO.

Reply:
I updated the PR to use `force_drop` from `opts` but it turns out that those opts are the `repo.config`. See https://github.com/elixir-ecto/ecto/blob/e58209ce7af4008d9015c3ff7084a20239103fbd/lib/mix/tasks/ecto.drop.ex#L74

How to proceed?

Reply:
Would this be Ok?
```elixir
case repo.__adapter__.storage_down(Keyword.merge([force_drop: opts[:force_drop]], repo.config)) do
```

That would even allow you to set this up in repo.config :)